File: src/main/java/com/ahold/technl/sandbox/exception/DeliveryException.java
package com.ahold.technl.sandbox.exception;

import lombok.Data;

@Data
public class DeliveryException extends RuntimeException{

    private int errorCode;
    private String errorMsg;

    public DeliveryException(int errorCode, String errorMsg) {
        super(errorMsg);
        this.errorCode = errorCode;
        this.errorMsg = errorMsg;
    }

    public DeliveryException(Throwable cause, int errorCode, String errorMsg) {
        super(cause);
        this.errorCode = errorCode;
        this.errorMsg = errorMsg;
    }
}

File: src/main/kotlin/com/ahold/technl/sandbox/exception/DeliveryException.kt
package com.ahold.technl.sandbox.exception

class DeliveryException : RuntimeException {
    val errorCode: Int
    val errorMsg: String

    constructor(errorCode: Int, errorMsg: String) : super(errorMsg) {
        this.errorCode = errorCode
        this.errorMsg = errorMsg
    }

    constructor(cause: Throwable?, errorCode: Int, errorMsg: String) : super(cause) {
        this.errorCode = errorCode
        this.errorMsg = errorMsg
    }
}